File: 1308/1308.py
class Solution:
    def running_total_for_different_genders(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.running_total_for_different_genders(root.left), self.running_total_for_different_genders(root.right))

File: 1308/1308.go
func runningTotalForDifferentGender(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := runningTotalForDifferentGender(root.Left)
	r := runningTotalForDifferentGender(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}